applications/crossbar/src/modules/cb_search.erl

```diff
@@ -125,6 +164,79 @@ validate_search(Context, Type, Query) ->
             Context1
     end.
 
+validate_search(Context, _Type, _Query, 'undefined') ->
+    cb_context:add_validation_error(
+      <<"v">>
+      ,<<"required">>
+      ,wh_json:from_list([{<<"message">>, <<"Search needs a value to search with">>}])
+      ,Context
+     );
+validate_search(Context, Type, Query, <<_/binary>> = Value) ->
+    search(Context, Type, Query, Value);
+validate_search(Context, Type, Query, Value) ->
+    case wh_util:is_true(Value) of
+        'true' -> validate_search(Context, Type, Query, <<>>);
+        'false' -> validate_search(Context, Type, Query, 'undefined')
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec validate_multi(cb_context:context(), api_binary()) -> cb_context:context().
+-spec validate_multi(cb_context:context(), ne_binary(), wh_proplist()) -> cb_context:context().
+validate_multi(Context, 'undefined') ->
+    cb_context:add_validation_error(
+        <<"t">>
+        ,<<"required">>
+        ,wh_json:from_list([{<<"message">>, <<"Search needs a document type to search on">>}])
+        ,Context
+    );
+validate_multi(Context, Type) ->
+    QS = cb_context:query_string(Context),
+    case wh_json:to_proplist(wh_json:delete_key(<<"t">>, QS)) of
+        [_|_]=Props -> validate_multi(Context, Type, Props);
+        _Other ->
+            cb_context:add_validation_error(
+                <<"multi">>
+                ,<<"required">>
+                ,wh_json:from_list([{<<"message">>, <<"Multi Search needs something to search">>}])
+                ,Context
+            )
+    end.
+
+validate_multi(Context, Type, Props) ->
+    QueryOptions = query_options(cb_context:account_db(Context)),
+    case validate_multi_fold(QueryOptions, Props) of
+        'ok' -> multi_search(Context, Type, Props);
+        Invalid ->
+            cb_context:add_validation_error(
+                <<"multi">>
+                ,<<"enum">>
+                ,wh_json:from_list([
+                    {<<"message">>, <<"Value not found in enumerated list of values">>}
+                    ,{<<"target">>, QueryOptions}
+                    ,{<<"cause">>, Invalid}
+                 ])
+                ,Context
+            )
+    end.
+
+-spec validate_multi_fold(wh_proplist(), wh_proplist()) -> 'ok' | ne_binary().
+validate_multi_fold(_Options, []) -> 'ok';
+validate_multi_fold(Options, [{Query, _}|Props]) ->
```

Comment:
In fact, you should probably just call into `validate_query/3` and check the resulting Context for error to see if you should end the loop